### DISCUSSION TITLE
Composite key on n:m relation table

### DISCUSSION THREAD

**Discussion by ff-anthony-soto** [prisma/prisma#1999] [Q&A] [answered]
``` 
model Muscle {
  id Int @id @default(autoincrement())
  name String @unique
  exercises MuscleExercise[]
}

model MuscleExercise {
  muscle  Muscle @unique @relation
  exercise Exercise @unique @relation 
  priority Int
}

model Exercise {
  id String @id @default(uuid())
  name String @unique
  muscles MuscleExercise[]
}
``` 

I need to simulate an n:m table (`MuscleExercise`), but when I'm running the studio I'm getting an error:

```
PANIC: Unable to resolve a primary identifier for model MuscleExercise.
at PrismaClientFetcher.request
```

What's the way to do a composite key between tables?

**Comment by janpio** [comment, marked as answer]
Should both `muscle` and `exercise` in `MuscleExercis` both be unique _individually_ (which is how it is modeled right now) or as a compound unique? If the latter, you should try `@@unique([muscle, execrise])` - that should be treated as the unique identifier for that model.

**Reply by ff-anthony-soto** [reply]
Exactly, it has to be a compound unique! Thanks!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
